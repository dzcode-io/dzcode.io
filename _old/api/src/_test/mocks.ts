/* eslint-disable camelcase */
import { GithubUser } from "src/github/types";

export const githubUserMock: GithubUser = {
  login: "ZibanPirate",
  id: 20110076,
  node_id: "MDQ6VXNlcjIwMTEwMDc2",
  avatar_url: "https://avatars.githubusercontent.com/u/20110076?v=4",
  gravatar_id: "",
  url: "https://api.github.com/users/ZibanPirate",
  html_url: "https://github.com/ZibanPirate",
  followers_url: "https://api.github.com/users/ZibanPirate/followers",
  following_url: "https://api.github.com/users/ZibanPirate/following{/other_user}",
  gists_url: "https://api.github.com/users/ZibanPirate/gists{/gist_id}",
  starred_url: "https://api.github.com/users/ZibanPirate/starred{/owner}{/repo}",
  subscriptions_url: "https://api.github.com/users/ZibanPirate/subscriptions",
  organizations_url: "https://api.github.com/users/ZibanPirate/orgs",
  repos_url: "https://api.github.com/users/ZibanPirate/repos",
  events_url: "https://api.github.com/users/ZibanPirate/events{/privacy}",
  received_events_url: "https://api.github.com/users/ZibanPirate/received_events",
  type: "User",
  site_admin: false,
  name: "Zakaria Mansouri",
  company: "@dzcode-io @avimedical",
  blog: "zak.dzcode.io",
  location: "Algeria",
  email: "",
  hireable: true,
  bio: "One-man-army lone programmer",
  twitter_username: "ZibanPirate",
  public_repos: 18,
  public_gists: 2,
  followers: 130,
  following: 92,
  created_at: "2016-06-23T12:41:14Z",
  updated_at: "2023-04-10T21:31:26Z",
};
