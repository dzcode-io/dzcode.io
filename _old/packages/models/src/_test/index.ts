import { OptionalPropertiesOf, RequiredPropertiesOf } from "@dzcode.io/utils/dist/ts";
import { ClassConstructor, plainToClass } from "class-transformer";
import { validateSync } from "class-validator";
import { BaseEntity } from "src/_base";

export type Cases<T extends object> = [
  [string, RequiredPropertiesOf<T>, false],
  [string, Required<T>, false],
  [string, Record<string, never>, boolean],
];

export const runDTOTestCases = <T extends BaseEntity>(
  entity: ClassConstructor<T>,
  requiredFields: RequiredPropertiesOf<T>,
  optionalFields: OptionalPropertiesOf<T>,
): void => {
  const hasNoRequiredFields = Object.keys(requiredFields).length === 0;
  const cases: Cases<T> = [
    ["should match snapshot when providing required fields only", requiredFields, false],
    [
      "should match snapshot when providing all fields",
      { ...requiredFields, ...optionalFields } as Required<T>,
      false,
    ],
    [
      hasNoRequiredFields
        ? "should match snapshot when providing an empty object"
        : "should show an error that matches snapshot when passing empty object",
      {},
      !hasNoRequiredFields,
    ],
  ];

  it.each(cases)("%s", (name, input, expectErrors) => {
    const output = plainToClass(entity, input);
    const errors = validateSync(output);

    expect(output).toMatchSnapshot("output");
    expect(errors).toMatchSnapshot("errors");
    if (expectErrors) {
      expect(errors.length).not.toBeLessThanOrEqual(0);
    } else {
      expect(errors.length).toBe(0);
    }
  });
};
