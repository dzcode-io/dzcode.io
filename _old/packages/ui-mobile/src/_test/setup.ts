jest.mock("react-native-reanimated", () => {
  // eslint-disable-next-line @typescript-eslint/no-var-requires
  const Reanimated = require("react-native-reanimated/mock");

  // The mock for `call` immediately calls the callback which is incorrect
  // So we override it with a no-op
  Reanimated.default.call = () => undefined;

  return Reanimated;
});

// Silence the warning: Animated: `useNativeDriver` is not supported because the native animated module is missing
jest.mock("react-native/Libraries/Animated/NativeAnimatedHelper");

jest.mock("@gorhom/bottom-sheet", () => {
  // eslint-disable-next-line @typescript-eslint/no-var-requires
  const react = require("react-native");
  return {
    __esModule: true,
    default: react.View,
    BottomSheetScrollView: react.ScrollView,
  };
});
