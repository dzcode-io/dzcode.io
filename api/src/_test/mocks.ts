import { GithubUser } from "../app/types/legacy";

export const githubUserMock: GithubUser = {
  avatar_url: "avatar_url",
  html_url: "html_url",
  id: 1,
  login: "login",
  type: "type",
};

export const githubUserMock2: GithubUser = {
  avatar_url: "avatar_url2",
  html_url: "html_url2",
  id: 2,
  login: "login2",
  type: "type2",
};

export const githubUserMock3: GithubUser = {
  avatar_url: "avatar_url3",
  html_url: "html_url3",
  id: 3,
  login: "login3",
  type: "type3",
};
