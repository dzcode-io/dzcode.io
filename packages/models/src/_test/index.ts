import {
  OptionalPropertiesOf,
  RequiredPropertiesOf,
} from "@dzcode.io/utils/dist/ts";
import { ClassConstructor, plainToClass } from "class-transformer";
import { validateSync } from "class-validator";
import { BaseEntity } from "../_base";

export type Cases<T extends object> = [
  {
    name: string;
    input: RequiredPropertiesOf<T>;
    expectErrors: false;
  },
  {
    name: string;
    input: Required<T>;
    expectErrors: false;
  },
  {
    name: string;
    input: Record<string, never>;
    expectErrors: boolean;
  }
];

export const runDTOTestCases = <T extends BaseEntity>(
  entity: ClassConstructor<T>,
  requiredFields: RequiredPropertiesOf<T>,
  optionalFields: OptionalPropertiesOf<T>
): void => {
  const hasNoRequiredFields = Object.keys(requiredFields).length === 0;
  const cases: Cases<T> = [
    {
      name: "should match snapshot when providing required fields only",
      input: requiredFields,
      expectErrors: false,
    },
    {
      name: "should match snapshot when providing all fields",
      input: { ...requiredFields, ...optionalFields } as Required<T>,
      expectErrors: false,
    },
    {
      name: hasNoRequiredFields
        ? "should match snapshot when providing an empty object"
        : "should show an error that matches snapshot when passing empty object",
      input: {},
      expectErrors: !hasNoRequiredFields,
    },
  ];

  it.each(cases)("$name", ({ input, expectErrors }) => {
    const output = plainToClass(entity, input);
    const errors = validateSync(output);

    expect(output).toMatchSnapshot("output");
    expect(errors).toMatchSnapshot("errors");
    if (expectErrors) {
      expect(errors.length).not.toBeLessThanOrEqual(0);
    } else {
      expect(errors.length).toBe(0);
    }
  });
};
