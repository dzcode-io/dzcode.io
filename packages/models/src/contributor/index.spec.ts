import { ContributorEntity } from ".";
import { runDTOTestCases } from "../_test";

runDTOTestCases(
  ContributorEntity,
  {
    avatarUrl: "https://avatars.githubusercontent.com/u/20110076?v=4",
    id: "20110076",
    username: "ZibanPirate",
  },
  {
    repositories: [],
  }
);
