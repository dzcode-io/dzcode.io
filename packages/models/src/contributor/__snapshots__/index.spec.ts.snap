// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot when providing all fields: errors 1`] = `Array []`;

exports[`should match snapshot when providing all fields: output 1`] = `
ContributorEntity {
  "avatarUrl": "https://avatars.githubusercontent.com/u/20110076?v=4",
  "id": "20110076",
  "repositories": Array [],
  "username": "ZibanPirate",
}
`;

exports[`should match snapshot when providing required fields only: errors 1`] = `Array []`;

exports[`should match snapshot when providing required fields only: output 1`] = `
ContributorEntity {
  "avatarUrl": "https://avatars.githubusercontent.com/u/20110076?v=4",
  "id": "20110076",
  "username": "ZibanPirate",
}
`;

exports[`should show an error that matches snapshot when passing empty object: errors 1`] = `
Array [
  ValidationError {
    "children": Array [],
    "constraints": Object {
      "isString": "id must be a string",
    },
    "property": "id",
    "target": ContributorEntity {},
    "value": undefined,
  },
  ValidationError {
    "children": Array [],
    "constraints": Object {
      "isString": "username must be a string",
    },
    "property": "username",
    "target": ContributorEntity {},
    "value": undefined,
  },
  ValidationError {
    "children": Array [],
    "constraints": Object {
      "isUrl": "avatarUrl must be an URL address",
    },
    "property": "avatarUrl",
    "target": ContributorEntity {},
    "value": undefined,
  },
]
`;

exports[`should show an error that matches snapshot when passing empty object: output 1`] = `ContributorEntity {}`;
